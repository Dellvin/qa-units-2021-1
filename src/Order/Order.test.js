jest.mock('../utils/getDate');

import React from 'react'
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Order from "./Order";
import {getDate} from "../utils/getDate";

Enzyme.configure({adapter: new Adapter()})

describe('Order', () => {

    beforeAll(() => getDate.mockReturnValue('26 декабря, пн, 2000 год'));

    afterAll(() => jest.clearAllMocks());

    it('undefined order.shops and order.date', () => {
        const order = {
            date: undefined,
            shop: undefined,
        };

        const output = shallow(
            <Order order={order}/>
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });

    it('Order props', () => {

        const order = {
            date: 127001,
            shop: 'ozon',
            items: ['1234', '2374123']
        };

        const output = shallow(
            <Order order={order}/>
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });

    it('check times', () => {
        expect(getDate).toHaveBeenCalledTimes(1);
    });

    it('undefined order.items', () => {
        const order = {
            date: 1541111800,
            shop: 'ozon',
            items: undefined,
        };

        const output = shallow(
            <Order order={order}/>
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });

    it('check times', () => {
        expect(getDate).toHaveBeenCalledTimes(2);
    });

    it('undefined order', () => {
        const order = undefined;

        const output = shallow(
            <Order order={order}/>
        );
        expect(output).toEqual({});
    });

    it('render no props', () => {
        const output = shallow(
            <Order/>
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });

    it('order.items empty', () => {
        const order = {
            date: 872341,
            shop: 'ozon',
            items: [],
        };

        const output = shallow(
            <Order order={order}/>
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });

    it('check times', () => {
        expect(getDate).toHaveBeenCalledTimes(3);
    });

    it('undefined order.date', () => {
        const order = {
            date: undefined,
        };

        const output = shallow(
            <Order order={order}/>
        );
      expect(output).toEqual({});
    });

    it('undefined order.shop', () => {
        const order = {
            shop: undefined,
        };

        const output = shallow(
            <Order order={order}/>
        );
      expect(output).toEqual({});
    });
});