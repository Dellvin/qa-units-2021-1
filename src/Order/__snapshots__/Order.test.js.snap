// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order Order props 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      ozon
    </span>
    <span
      id="date"
    >
      26 декабря, пн, 2000 год
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      1234
    </div>
    <div
      className="Order-item"
      key="1"
    >
      2374123
    </div>
  </div>
</div>
`;

exports[`Order order.items empty 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      ozon
    </span>
    <span
      id="date"
    >
      26 декабря, пн, 2000 год
    </span>
  </div>
  <div
    className="Order-items"
  />
</div>
`;

exports[`Order render no props 1`] = `""`;

exports[`Order undefined order.items 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      ozon
    </span>
    <span
      id="date"
    >
      26 декабря, пн, 2000 год
    </span>
  </div>
  <div
    className="Order-items"
  />
</div>
`;

exports[`Order undefined order.shops and order.date 1`] = `""`;
